bitcoind-rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindConfig.scala

```diff
@@ -224,13 +242,14 @@ abstract class BitcoindConfig extends BitcoinSLogger {
   /** Creates a new config with the given keys and values appended */
   def withOption(key: String, value: String): BitcoindConfig = {
     val ourLines = this.lines
-    new BitcoindConfig {
-
-      def lines: Seq[String] = {
-        val newLine = s"$key=$value"
-        newLine +: ourLines
-      }
-    }
+    val newLine = s"$key=$value"
+    val lines = newLine +: ourLines
+    val newConfig = BitcoindConfig(lines, datadir)
+    logger.debug(
+      s"Appending new config with $key=$value to datadir=${datadir.getAbsolutePath}")
+    BitcoindConfig.writeConfigToFile(newConfig, datadir)
```

Comment:
Not sure how I feel about a seemingly innocent function performing a IO side effect, without that being explicitly communicated. Same thing applies to the places above we're also calling `BitcoindConfig.writeConfigToFile`. Would be bad if you're playing around with the config class, and without knowing it you're overwriting your `$HOME/.bitcoin` directory

Reply:
So I agree that is a downside, but the other downside is having divergence between what is on disk and what is represented in memory. This is a prepend only operation, so it does preserve the contents of the file before this operation is called. 

I guess this is more theoretical, but do we want to have the ability to diverge from what is on disk and what is actually contained in this case class? I would tend to say no.